src/SignalR/common/Protocols.MessagePack/src/Protocol/MessagePackHubProtocol.cs

```diff
@@ -71,13 +73,14 @@ private void SetupResolver(MessagePackHubProtocolOptions options)
                 // check if the user customized the resolvers
                 if (options.FormatterResolvers[i] != SignalRResolver.Resolvers[i])
                 {
-                    _resolver = new CombinedResolvers(options.FormatterResolvers);
+                    var resolver = CompositeResolver.Create(Array.Empty<IMessagePackFormatter>(), (IReadOnlyList<IFormatterResolver>)options.FormatterResolvers);
+                    _msgPackSerializerOptions = MessagePackSerializerOptions.Standard.WithResolver(resolver);
```

Comment:
If app controlled data is ever sent over this, we should expose a way for the app owner to specify the `MessagePackSerializerOptions` instance to use (including observing the one you prepared so they can base their version on yours to avoid breaking you).

Reply:
sounds fair, what's your idea ?
I though about changing the `IOptions< MessagePackHubProtocolOptions >` exposed list of resolver since it's only used for that. It would let consumer manipulate the actual instance that's gonna be used

but it's might be a bit to violent ^^ or me going to break something i ddi not knew about ;)

Reply:
Yes, changing/adding to the `MessagePackHubProtocolOptions` type looks like it would make sense. The most natural thing from a messagepack v2 perspective would be to replace the `FormatterResolvers` property with a `MessagePackSerializerOptions` property. Or perhaps to remove the `MessagePackHubProtocolOptions` altogether in favor of just passing around `MessagePackSerializerOptions` directly.

Reply:
that conforts me in doing so ^^
will try to take a look ASAP as i'm off this weekend and hope to at least check it before

Reply:
I agree we should take a look at changing the options, but lets file an issue and do it as a separate work item. Changing APIs will require API reviews etc.

Reply:
fine for me ;)
i let you create the issue @BrennanConroy ? as you probably have a better understanding of how to describe the checklist ?
Also might be a a good idea to include this change in a the announcement issue for both major changes then
not sure if there a tag that should be added indicating that it should be "shown/published" whatever so that when the `previewxxx` + major release note it's easy to "get them all"

Reply:
Filed https://github.com/dotnet/aspnetcore/issues/18569